crates/bevy_input/src/gamepad/system.rs

```diff
@@ -0,0 +1,120 @@
+use crate::{
+    gamepad::{
+        GamepadAxis, GamepadButton, GamepadEvent, GamepadEventRaw, GamepadEventType,
+        GamepadSettings, Gamepads, ALL_AXIS_TYPES, ALL_BUTTON_TYPES,
+    },
+    Axis, Input,
+};
+use bevy_app::{EventReader, EventWriter};
+use bevy_ecs::system::{Res, ResMut};
+use bevy_utils::tracing::info;
+
+/// Monitors gamepad connection and disconnection events and updates the [`Gamepads`] resource accordingly.
+///
+/// ## Schedule
+///
+/// By default, this system runs during `CoreStage::PreUpdate` when added via the [`InputPlugin`](crate::InputPlugin).
+///
+/// ## Note
+///
+/// Whenever a [`Gamepad`](crate::gamepad::Gamepad) connects or disconnects, an information gets printed to the console using the [`info!`] macro.
+pub fn gamepad_connection_system(
+    mut gamepads: ResMut<Gamepads>,
+    mut gamepad_event: EventReader<GamepadEvent>,
+) {
+    for event in gamepad_event.iter() {
+        match event.event_type {
+            GamepadEventType::Connected => {
+                gamepads.register(event.gamepad);
+                info!("{:?} Connected", event.gamepad);
+            }
+            GamepadEventType::Disconnected => {
+                gamepads.deregister(&event.gamepad);
+                info!("{:?} Disconnected", event.gamepad);
+            }
+            _ => (),
+        }
+    }
+}
+
+/// Modifies the gamepad resources and sends out gamepad events.
+///
+/// The resources [`Input<GamepadButton>`], [`Axis<GamepadAxis>`], and [`Axis<GamepadButton>`] are updated
+/// and the [`GamepadEvent`]s are sent according to the received [`GamepadEventRaw`]s.
+///
+/// ## Differences
+///
+/// The main difference between the events and the resources is that the latter have convenient functions
```

Comment:
I think it's more that with the input you can check whether specific buttons are pressed, rather than reading the events one at a time.